pkg/oc/cli/cmd/rollout/latest.go

```diff
@@ -175,17 +238,70 @@ func (o RolloutLatestOptions) RunRolloutLatest() error {
 		}
 
 		if err != nil {
-			return err
+			return nil, err
 		}
+	}
+
+	return dc, nil
+}
 
+// RunRolloutLatest runs the latest rollouts.
+func (o RolloutLatestOptions) RunRolloutLatest() error {
+	var revision string
+	info := o.infos[0]
+	switch obj := info.Object.(type) {
+	case *deployapi.DeploymentConfig:
+		dc, err := o.rolloutDeploymentConfig(obj)
+		if err != nil {
+			return err
+		}
+		revision = fmt.Sprintf("%d", dc.Status.LatestVersion)
 		info.Refresh(dc, true)
+	case *kextensions.Deployment:
+		updatedObj, err := o.rolloutFromAnnotation(obj, func(in runtime.Object) (runtime.Object, error) {
```

Comment:
@mfojtik I am not sure why to give update function as an argument to the inner function just to call it while doing return. Wouldn't it have the same effect if you would do it right after calling the function on the result and there would be one less argument and lambda.

Reply:
yeah, i regretted doing it this way :)